x/operator/keeper/abci.go

```diff
@@ -0,0 +1,135 @@
+package keeper
```

Comment:
Are we updating the shares at each block? The dogfood module is designed to handle this only on a per-epoch basis to lower the amount of processing required every 3 seconds.

Reply:
Yeah, the shares will be updated each block. I saw the dogfood module is designed to update states only on a per-epoch basis. In terms of the functionality, I don't think it will have any impact on dogfood. In the future, if performance issues are considered, epoch may be introduced into the operator module.

Reply:
Has this approach been discussed with RJ? Updating each block is very cost heavy.

Reply:
Yeah, this has been discussed with RJ. The first version is just a version to implement the basic features. The epoch will be addressed in the next version. 
In the current implementation, the assets' amount and share of stakers and operators will be updated immediately when the related delegation/undelegation, slash, opt-in/out and price change occur.  The share update caused by the price change will be handled on a per-epoch basis after the code is implemented with Oracle hook.  As for the other situation, we might keep the update of assets' amount unchanged but change the share update to be on a per-epoch basis.